From 493c803f2cceb323d30fc15c2c72af5e1bdb3fa3 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:34 -0400
Subject: [tools] perf/completion: Strip function_exists ()

Message-id: <1408462094-14194-106-git-send-email-jolsa@redhat.com>
Patchwork-id: 88078
O-Subject: [PATCH RHEL7.1 BZ1131394 105/325] perf completion: Strip function_exists ()
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 4685a6cfaeac6d2fe8ed10a9aa0cbff5026529cb
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Thu Jul 4 18:11:29 2013 +0530

description
===========
Use "type" to check existence consistently.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/bash_completion b/tools/perf/bash_completion
index d2598be..35fdda1 100644
--- a/tools/perf/bash_completion
+++ b/tools/perf/bash_completion
@@ -1,12 +1,6 @@
 # perf completion
 
-function_exists()
-{
- declare -F $1 > /dev/null
- return $?
-}
-
-function_exists __ltrim_colon_completions ||
+type __ltrim_colon_completions &>/dev/null ||
 __ltrim_colon_completions()
 {
  if [[ "$1" == *:* && "$COMP_WORDBREAKS" == *:* ]]; then
-- 
1.7.1